src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.CommandResult;
+import duke.command.ExitCommand;
+import duke.ui.TextUi;
+
+public class Duke {
+    // Global Variables
+    private static final String FILENAME = "task.txt";
+
+    private TaskList taskList;
+    private Storage fh;
+    private TextUi ui;
+    private Parser parser;
+
+
+    /**
+     * The main entry point to the application
+     *
+     * Creates an instance of Duke and starts the application
+     */
+    public static void main(String[] args) {
+        new Duke().run();
+    }
+
+    /**
+     * Starts the main logic of the program
+     *
+     * Instantiates TextUi, Storage, TaskList, and Parser objects and starts input loop
+     */
+    public void run() {
+        // Setting up the goods
+        ui = new TextUi();
+        fh = new Storage(FILENAME);
+        taskList = new TaskList(fh.importTasks());
+        parser = new Parser();
+        parser.setTaskList(taskList);
+
+        ui.printIntro();
+
+        // Starting input loop
+        while (true) {
+            String input = ui.getInput();
+            if (processInput(input)) {
```

Comment:
Since `processInput()` returns a `Boolean` value, perhaps you can use another function with a name like `isInputExitCommand()` and separate the logic for processing the input for other commands?